app/graphql/mutations/vote_for_survey.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: false
+
+module Mutations
+  class VoteForSurvey < BaseMutation
+    argument :increase_id, ID, required: false
+    argument :decrease_id, ID, required: false
+
+    type Types::StatusType
+
+    def resolve(increase_id: nil, decrease_id: nil)
+      return error_status("Access not permitted: missing role") unless context[:current_user].app_role?
+      return error_status("Missing ResponseOptionID as increase_id or decrease_id") if increase_id.blank? && decrease_id.blank?
+
+      begin
+        increase_reponse = Survey::ResponseOption.find_by(id: increase_id) if increase_id.present?
+        increase_reponse.increment!(:votes_count) if increase_reponse.present?
+
+        decrease_reponse = Survey::ResponseOption.find_by(id: decrease_id) if decrease_id.present?
+        decrease_reponse.decrement!(:votes_count) if decrease_reponse.present?
+
+        OpenStruct.new(id: nil, status: "VoteForSurvey successful ", status_code: 200)
```

Comment:
hier vllt. was eindeutigeres? angelehnt an visibility, wo steht `"visibility set to #{visible}"` könnt ich mir hier vorstellen z.B.: `"voted successfully, increased: #{increase_id} decreased: #{decrease_id}"`?

Reply:
angepasst